pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -364,6 +368,11 @@ func (r *Range) Destroy() {
 	r.alloc.Destroy()
 }
 
+// EnableMetrics enables metrics recording.
+func (r *Range) EnableMetrics() {
+	r.metrics = &metricsRecorder{}
```

Comment:
should the metrics `registerMetrics()`  be initialized here  now?

Reply:
Thank you. I fixed it.

https://github.com/kubernetes/kubernetes/compare/f2f925a67f91f496e3875b8a8131f7eb243e352e..187af7781a4d6a308ea7f73dd22d435575873396#diff-3f7680212eb37166b66a0f052153bb765c0d4b141af958d5b92543dda628d35fR373